bundles/org.openhab.binding.electroluxappliances/src/main/java/org/openhab/binding/electroluxappliances/internal/handler/ElectroluxWashingMachineHandler.java

```diff
@@ -0,0 +1,173 @@
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.electroluxappliances.internal.handler;
+
+import static org.openhab.binding.electroluxappliances.internal.ElectroluxAppliancesBindingConstants.*;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.electroluxappliances.internal.ElectroluxAppliancesConfiguration;
+import org.openhab.binding.electroluxappliances.internal.dto.ApplianceDTO;
+import org.openhab.binding.electroluxappliances.internal.dto.WashingMachineStateDTO;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.OpenClosedType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.library.unit.Units;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.Channel;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.State;
+import org.openhab.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link ElectroluxWashingMachineHandler} is responsible for handling commands and status updates for
+ * Electrolux washing machines.
+ *
+ *
+ * @author Jan Gustafsson - Initial contribution
+ */
+@NonNullByDefault
+public class ElectroluxWashingMachineHandler extends ElectroluxAppliancesHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(ElectroluxWashingMachineHandler.class);
+
+    private ElectroluxAppliancesConfiguration config = new ElectroluxAppliancesConfiguration();
+
+    public ElectroluxWashingMachineHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("Command received: {} on channelID: {}", command, channelUID);
+        if (CHANNEL_STATUS.equals(channelUID.getId()) || command instanceof RefreshType) {
+            super.handleCommand(channelUID, command);
+        }
+    }
+
+    @Override
+    public void update(@Nullable ApplianceDTO dto) {
+        if (dto != null) {
+            // Update all channels from the updated data
+            getThing().getChannels().stream().map(Channel::getUID).filter(channelUID -> isLinked(channelUID))
+                    .forEach(channelUID -> {
+                        State state = getValue(channelUID.getId(), dto);
+                        logger.trace("Channel: {}, State: {}", channelUID, state);
+                        updateState(channelUID, state);
+                    });
+            updateStatus(ThingStatus.ONLINE);
+        }
+    }
+
+    private State getValue(String channelId, ApplianceDTO dto) {
+        switch (channelId) {
```

Comment:
Use temp cache to prevent to do all the plumbing for every channel again. Similar can also be applied to the air-purifier handler.
```suggestion
        var reported = ((WashingMachineStateDTO) dto.getApplianceState()).getProperties().getReported();
        switch (channelId) {
```

Reply:
Fixed!